siriuspy/siriuspy/devices/ids.py

```diff
@@ -106,6 +108,249 @@ def wait_move(self):
             _time.sleep(APU._MOVECHECK_SLEEP)
 
 
+class EPU(_Device):
+    """."""
+
+    class DEVICES:
+        """."""
+        EPU50_10SB = 'SI-10SB:ID-EPU50'
+        ALL = (EPU50_10SB, )
+
+    _idparam_fields = (
+        'PERIOD',  # [mm]
+        'PHASE_MIN',  # [mm]
+        'PHASE_MAX',  # [mm]
+        'PHASE_PARK',  # [mm]
+        'GAP_MIN',   # [mm]
+        'GAP_MAX',  # [mm]
+        'GAP_PARK',  # [mm]
+        )
+
+    _dev2params = {
+        DEVICES.EPU50_10SB :
+            _get_namedtuple('IDParameters',
+                _idparam_fields, (50.0, -50.0/4, 50/4, 0, 22.0, 300.0, 300.0)),
+        }
+                
+    _default_timeout = 5  # [s]
+
+    _properties = (
+        'BeamLineCtrlEnbl-Sel', 'BeamLineCtrlEnbl-Sts',
+        'EnblPwrAll-Cmd', 'PwrPhase-Mon', 'PwrGap-Mon',
+        'EnblAndReleasePhase-Sel', 'EnblAndReleasePhase-Sts',
+        'AllowedToChangePhase-Mon',
+        'EnblAndReleaseGap-Sel', 'EnblAndReleaseGap-Sts',
+        'AllowedToChangeGap-Mon',
+        'Phase-SP', 'Phase-RB', 'Phase-Mon',
+        'PhaseSpeed-SP', 'PhaseSpeed-RB', 'PhaseSpeed-Mon',
+        'Gap-SP', 'Gap-RB', 'Gap-Mon',
+        'GapSpeed-SP', 'GapSpeed-RB', 'GapSpeed-Mon',
+        'Stop-Cmd', 'Moving-Mon', 'IsBusy-Mon',
+    )
+
+    def __init__(self, devname):
+        """."""
+        devname = _SiriusPVName(devname)
+
+        # check if device exists
+        if devname not in EPU.DEVICES.ALL:
+            raise NotImplementedError(devname)
+
+        # call base class constructor
+        super().__init__(devname, properties=EPU._properties, auto_mon=True)
+
+    @property
+    def parameters(self):
+        """."""
+        return EPU._dev2params[self.devname]
+
+    @property
+    def speed_phase(self):
+        """Phase speed [mm/s]."""
+        return self['PhaseSpeed-Mon']
```

Comment:
Maybe use `'PhaseSpeed-RB'`here and create a `speed_phase_mon` property for `'PhaseSpeed-Mon'`

Reply:
thanks, @anacso17 . fixed.